### PR TITLE
FIX: [amcs] Handle 3D -> 2D Mono

### PR DESCRIPTION
**Pull request by koying**

### CODE REVIEW

**Comment by koying** [comment]
jenkins build this please

**Comment by da-anda** [comment]
tested on the SHIELD and fixes the issue only partially. 3D movies are now always 2D, regardless of setting ;)

**Comment by da-anda** [comment]
tested some more and have a better picture of the bug now. It seems as if the surface won't scale back to regular size once you turned on 2D rendering. So if you start in 3D mode, everything is fine, if you then switch to 2D mode, everything is still fine, but if you then switch back to 3D, picture will stay 2D.

File: xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp [hunk 1]
```diff
@@ -143,6 +143,11 @@ bool CRendererMediaCodecSurface::RenderHook(int index)
         srcRect.x2 *= 2.0;
       break;
 
+      case RENDER_STEREO_MODE_MONO:
+        dstRect.y2 = dstRect.y2 * (dstRect.y2 / m_sourceRect.y2);
+        dstRect.x2 = dstRect.x2 * (dstRect.x2 / m_sourceRect.x2);
```
Comment (FrankHeimes) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp]: This is: `dstRect.x2 *= dstRect.x2 / m_sourceRect.x2`.
If `dstRect` is narrow and `m_sourceRect` is wide, then `dstRect` becomes more narrow?
All instructions in this method silently assume that `.x1 = 0` and `.y1 = 0`. Is this correct?

Comment (koying) [inline xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp]: This assumes Mono uses top image of TAB and left image of SBS, indeed.
Not sure about the "narrow" comment. I don't know of a use case where TAB/SBS is not full resolution, really.

**Comment by fritsch** [comment]
The second commit is not on topic but looks correct.

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Pull request merged**